File: src/backend/laravel-app/app/Models/Regle.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;


/**
 * @OA\Schema(
 *     required={"libelle", "reglementInterieurId"},
 *     @OA\Xml(name="Regle"),
 *     @OA\Property(property="id", type="integer", readOnly=true, example="1"),
 *     @OA\Property(property="reglementInterieurId", type="integer", example=1),
 *     @OA\Property(property="libelle", type="string", readOnly=true, description="Une regle"),
 *     @OA\Property(property="reglementInterieur", ref="#/components/schemas/ReglementInterieur"),
 *     @OA\Property(property="created_at", ref="#/components/schemas/BaseModel/properties/created_at"),
 *     @OA\Property(property="updated_at", ref="#/components/schemas/BaseModel/properties/updated_at"),
 *     @OA\Property(property="deleted_at", ref="#/components/schemas/BaseModel/properties/deleted_at")
 * )
 *
 * @method static create(array $array)
 * @method static findOrFail($id)
 * @method static paginate(int $int)
 *
 * Class Regle
 *
 * @package App\Models
 */
class Regle extends Model
{
    use HasFactory;

    protected $fillable = [
        'libelle',
        'reglementInterieurId',
        'reglementInterieur',
    ];

    public function reglementInterieur()
    {
        return $this->belongsTo(ReglementInterieur::class, 'reglementInterieurId');
    }

    public function fautes()
    {
        return $this->hasMany(Faute::class, 'regleId');
    }

}

File: src/frontend/mobile/flutter_app/lib/common/models/regle.dart
class Regle {
  final int id;
  final String libelle;
  final int reglementInterieurId;
  final String createdAt;
  final String updatedAt;

  Regle({
    required this.id,
    required this.libelle,
    required this.reglementInterieurId,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Regle.fromJson(Map<String, dynamic> json) => Regle(
      id: json['id'],
      libelle: json['libelle'],
      reglementInterieurId: json['reglementInterieurId'],
      createdAt: json['created_at'],
      updatedAt: json['updated_at']);
}
